### PR TITLE
Add std namespace prefix in C++ code

### PR DESCRIPTION
**Pull request by beenje**
Allow to compile with `TANGO_USE_USING_NAMESPACE=OFF` in `cppTango`.

Add `std::` prefix to:

- cout
- cerr
- endl
- ends
- vector
- string
- bad_alloc
- numeric_limits
- max
- auto_ptr

Should `std::auto_ptr` be replaced with `unique_pointer` instead?

### CODE REVIEW

**Comment by ajoubertza** [comment]
Hi @beenje

Welcome to PyTango!  Thanks for the PR.  It seems like a good idea.  Looking at the related tickets, were you trying to compile PyTango on MacOS?  That would be nice.

> Should `std::auto_ptr` be replaced with `unique_pointer` instead?

I've never noticed that before, but yes, it looks like the intention is to use `unique_pointer` everywhere except `ext/defs.h`, so please fix it.

File: ext/exception.cpp [hunk 1]
```diff
@@ -58,7 +58,7 @@ namespace Tango
 
 void sequencePyDevError_2_DevErrorList(PyObject *value, Tango::DevErrorList &del)
 {
-    long len = max((int)PySequence_Size(value), 0);
+    long len = std::max((int)PySequence_Size(value), 0);
```
Comment (ajoubertza) [inline ext/exception.cpp]: I see the AppVeyor build is failing [here](https://ci.appveyor.com/project/tiagocoutinho/pytango/builds/36759949/job/ly1bhcvmwlln2dhx?fullLog=true#L994) - the Visual Studio compiler doesn't like this.  Any idea why?  Can we leave this one out, or do we need an `#ifdef`?

Comment (beenje) [reply ext/exception.cpp]: It's an old answer but from https://stackoverflow.com/a/2789509, it looks like we need to use `#define NOMINMAX` for windows.
I'll give it a try.

Comment (beenje) [reply ext/exception.cpp]: I pushed a new commit that should fix the build on windows.

**Comment by beenje** [comment]
I tried to replace the `std::auto_ptr` with `unique_ptr` but the build fails.
```
    /tmp/pip-req-build-f94gz_85/ext/group.cpp:200:41:   required from here
    /opt/conda/envs/tango/include/boost/python/converter/as_to_python_function.hpp:40:33: error: use of deleted function 'std::unique_ptr<_Tp, _Dp>::unique_ptr(const std::unique_ptr<_Tp, _Dp>&) [with _Tp = Tango::Group; _Dp = std::default_delete<Tango::Group>]'
```

I kept `auto_ptr` in `ext/group.cpp` to test but it still fails with `unique_ptr` in `ext/server/device_class.cpp`:
```
    /tmp/pip-req-build-uppobmhw/ext/server/device_class.cpp:470:97:   required from here
    /opt/conda/envs/tango/include/boost/python/converter/implicit.hpp:37:9: error: no matching function for call to 'std::unique_ptr<CppDeviceClass>::unique_ptr(std::unique_ptr<CppDeviceClassWrap>&)'
       37 |         new (storage) Target(get_source());
          |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    In file included from /opt/conda/envs/tango/x86_64-conda-linux-gnu/include/c++/9.3.0/memory:80,
                     from /opt/conda/envs/tango/include/boost/function/function_base.hpp:16,
                     from /opt/conda/envs/tango/include/boost/function/detail/prologue.hpp:17,
                     from /opt/conda/envs/tango/include/boost/function/function_template.hpp:13,
                     from /opt/conda/envs/tango/include/boost/function/detail/maybe_include.hpp:15,
                     from /opt/conda/envs/tango/include/boost/function/function0.hpp:11,
                     from /opt/conda/envs/tango/include/boost/python/errors.hpp:13,
                     from /opt/conda/envs/tango/include/boost/python/handle.hpp:11,
                     from /opt/conda/envs/tango/include/boost/python/args_fwd.hpp:10,
                     from /opt/conda/envs/tango/include/boost/python/args.hpp:10,
                     from /opt/conda/envs/tango/include/boost/python.hpp:11,
                     from /tmp/pip-req-build-uppobmhw/ext/precompiled_header.hpp:20,
                     from /tmp/pip-req-build-uppobmhw/ext/server/device_class.cpp:12:
    /opt/conda/envs/tango/x86_64-conda-linux-gnu/include/c++/9.3.0/bits/unique_ptr.h:281:2: note: candidate: 'template<class _Up, class> std::unique_ptr<_Tp, _Dp>::unique_ptr(std::auto_ptr<_Up>&&)'
      281 |  unique_ptr(auto_ptr<_Up>&& __u) noexcept;
          |  ^~~~~~~~~~
    /opt/conda/envs/tango/x86_64-conda-linux-gnu/include/c++/9.3.0/bits/unique_ptr.h:281:2: note:   template argument deduction/substitution failed:
    In file included from /opt/conda/envs/tango/include/boost/python/implicit.hpp:10,
                     from /opt/conda/envs/tango/include/boost/python.hpp:33,
                     from /tmp/pip-req-build-uppobmhw/ext/precompiled_header.hpp:20,
                     from /tmp/pip-req-build-uppobmhw/ext/server/device_class.cpp:12:
    /opt/conda/envs/tango/include/boost/python/converter/implicit.hpp:37:9: note:   'std::unique_ptr<CppDeviceClassWrap>' is not derived from 'std::auto_ptr<T>'
       37 |         new (storage) Target(get_source());
```

Any idea?

**Comment by ajoubertza** [comment]
> I tried to replace the `std::auto_ptr` with `unique_ptr` but the build fails.
...
> Any idea?

No, I'm not sure why.  It looks like the boost library is still using `auto_ptr` for those conversions.  Maybe that's why those bits of the PyTango code don't use the `unique_ptr` typedef.    My C++ is very rusty, so I had a look [here](https://stackoverflow.com/questions/3451099/stdauto-ptr-to-stdunique-ptr) - the two types are not 100% compatible.

I'm happy if we just leave the `std::auto_ptr`s as they are.  Changing to `std::unique_ptr` is outside the scope of this PR.

File: ext/exception.cpp [hunk 2]
```diff
@@ -58,7 +58,7 @@ namespace Tango
 
 void sequencePyDevError_2_DevErrorList(PyObject *value, Tango::DevErrorList &del)
 {
-    long len = std::max((int)PySequence_Size(value), 0);
+    long len = (std::max)((int)PySequence_Size(value), 0);
```
Comment (ajoubertza) [inline ext/exception.cpp:61]: Nice.  I prefer this approach to messing with the include files.

**Review by ajoubertza — APPROVED** [review]
Looks good.

**Comment by ajoubertza** [comment]
@beenje Is there anything else you still want to add to this PR, or can I merge it?

**Comment by beenje** [comment]
If we don't touch the `auto_ptr`, I don't have anything else to add. It should be good to merge as is.

**Pull request merged**